                            Assignment:Estimation Confidence Intervals

In [17]:
# A:- Build 99% Confidence Interval Using Sample Standard Deviation

import scipy.stats as stats

# Sample Data
sample_data=[1.13, 1.55, 1.43, 0.92, 1.25, 1.36, 1.32, 0.85, 1.07, 1.48, 1.20, 1.33, 1.18, 1.22, 1.29]
sample_data

[1.13,
 1.55,
 1.43,
 0.92,
 1.25,
 1.36,
 1.32,
 0.85,
 1.07,
 1.48,
 1.2,
 1.33,
 1.18,
 1.22,
 1.29]

In [18]:
# Calculate sample statistics
sample_mean= sum(sample_data)/len(sample_data)
print("Mean of Sample data: ",sample_mean)
sample_std= stats.tstd(sample_data)  # using sample standard deviation
print("Standard deviation of Sample Data: ", sample_std)
sample_size= len(sample_data)
print("Sample Size: ", sample_size)

Mean of Sample data:  1.2386666666666666
Standard deviation of Sample Data:  0.19316412956959936
Sample Size:  15


In [19]:
# Confidence level
confidence_level = 0.99




In [20]:
# T-score for 99% confidence level and (n-1) degrees of freedom
t_score = stats.t.ppf((1 + confidence_level) / 2, df=sample_size-1)
print("T-score for 99% confidence level and (n-1) degrees of freedom: ", t_score)


T-score for 99% confidence level and (n-1) degrees of freedom:  2.97684273411266


In [21]:
# Margin of error
margin_of_error = t_score * (sample_std / (sample_size ** 0.5))
margin_of_error

0.1484693282152996

In [22]:
# confidence Level
confidence_level = (sample_mean - margin_of_error,sample_mean + margin_of_error)
print("Confidence Interval (Population Mean - Unknown Std Dev):0" , confidence_level)

Confidence Interval (Population Mean - Unknown Std Dev):0 (1.090197338451367, 1.3871359948819662)


In [23]:
# B:- Build 99% Confidence Interval Using Known Population Standard Deviation

import scipy.stats as stats

# Sample Data
sample_data=[1.13, 1.55, 1.43, 0.92, 1.25, 1.36, 1.32, 0.85, 1.07, 1.48, 1.20, 1.33, 1.18, 1.22, 1.29]

# Calculate sample statistics
sample_mean= sum(sample_data)/len(sample_data)
print("Mean of Sample data: ",sample_mean)
sample_std= 0.2  # using given sample standard deviation
print("Standard deviation of Sample Data: ", sample_std)
sample_size= len(sample_data)
print("Sample Size: ", sample_size)

# Confidence level
confidence_level = 0.99

# Z-score for 99% confidence level
z_score = stats.norm.ppf((1 + confidence_level) / 2)
print("Z-score for 99% confidence level: ", z_score)

# Margin of error
margin_of_error = t_score * (sample_std / (sample_size ** 0.5))
print('Margin of Error: ',margin_of_error)

# confidence level
confidence_level = (sample_mean - margin_of_error,sample_mean + margin_of_error)
print("Confidence Interval (Population Mean - Std Dev):" , confidence_level)

Mean of Sample data:  1.2386666666666666
Standard deviation of Sample Data:  0.2
Sample Size:  15
Z-score for 99% confidence level:  2.5758293035489004
Margin of Error:  0.15372349777995847
Confidence Interval (Population Mean - Std Dev): (1.0849431688867082, 1.392390164446625)


Observations:

A. Confidence Interval Using Sample Standard Deviation (t-distribution):

A sample of 15 observations was analyzed.

The sample mean is approximately 1.239.

The sample standard deviation is approximately 0.193.

Using a 99% confidence level and degrees of freedom (n-1 = 14), the t-score is 2.977.

The margin of error is calculated as 0.148.

Therefore, the 99% confidence interval for the population mean (with unknown population standard deviation) is approximately (1.090, 1.387).

B. Confidence Interval Using Known Population Standard Deviation (z-distribution):

The same sample mean of 1.239 is used, but the population standard deviation is assumed to be known and fixed at 0.2.

For a 99% confidence level, the z-score is 2.576.

Using this, the margin of error is 0.154, and the confidence interval becomes approximately (1.085, 1.392).

Conclusion:
Both methods produce similar but not identical confidence intervals for the population mean. When the population standard deviation is unknown, the t-distribution is more appropriate due to increased uncertainty, leading to a slightly wider interval. In contrast, using a known population standard deviation and the z-distribution results in a narrower interval. This exercise highlights the importance of correctly choosing the distribution based on available data and assumptions. In both cases, we are 99% confident that the true population mean lies within the respective intervals.

